paasta_tools/setup_kubernetes_job.py

```diff
@@ -254,6 +264,39 @@ def setup_kube_deployments(
         else (_, None)
         for _, service_instance in service_instance_configs_list
     ]
+
+    def sort_key(ok_app: Tuple[bool, Optional[Application]]) -> int:
+        """This will return 1 if the desired deployment_version matches an existing deployment_version, and 0
+        otherwise. This will cause applications that need a new deployment_version to be handled first.
```

Comment:
on another read: the `0 otherwise` bit isn't 100% accurate since we return 2 for broken apps